tests/test_reflection.py

```diff
@@ -33,3 +37,179 @@ def test_dialect(self):
             f"mysql+aoceanbase://{user}:{password}@{uri}/{db_name}?charset=utf8mb4"
         )
         self.engine = create_async_engine(connection_str)
+
+    def test_parse_constraints_with_string_spec(self):
+        """Test that _parse_constraints handles string spec gracefully without crashing."""
+        from pyobvector.schema.reflection import OceanBaseTableDefinitionParser
+        
+        # Create a mock parser class to test our specific method
+        class MockParser(OceanBaseTableDefinitionParser):
```

Comment:
I prefer to not mock anything here, the GitHub workflow will start a OceanBase server for testing, you can just use it with a specific SQL to test the new parser.